Content/Items/Misc/Weapons.Earthduster.cs

```diff
@@ -92,437 +106,1319 @@ public override void AddRecipes()
+			cache.Add(Projectile.Center + Projectile.velocity);
+
+			while (cache.Count > 8)
 			{
 				cache.RemoveAt(0);
 			}
 		}
 
 		private void ManageTrail()
 		{
-
 			if (trail is null || trail.IsDisposed)
-				trail = new Trail(Main.instance.GraphicsDevice, 33, new NoTip(), factor => 28 * (1 - Progress), factor => trailColorOutline);
-
-			if (trail2 is null || trail2.IsDisposed)
-				trail2 = new Trail(Main.instance.GraphicsDevice, 33, new NoTip(), factor => 10 * (1 - Progress), factor => trailColor);
-			float nextplace = 33f / 32f;
-			var offset = new Vector2((float)Math.Sin(nextplace), (float)Math.Cos(nextplace));
-			offset *= Radius;
+				trail = new Trail(Main.instance.GraphicsDevice, 8, new NoTip(), factor => 6f, factor => Colors["TrailColor"] * factor.X * FadeOut());
 
 			trail.Positions = cache.ToArray();
-			trail.NextPosition = Projectile.Center + offset;
+			trail.NextPosition = Projectile.Center;
+
+			if (trail2 is null || trail.IsDisposed)
+				trail2 = new Trail(Main.instance.GraphicsDevice, 8, new NoTip(), factor => 2.5f, factor => Color.Lerp(Colors["TrailInsideColor"], Colors["TrailColor"], 1f - factor.X) * factor.X * FadeOut());
 
 			trail2.Positions = cache.ToArray();
-			trail2.NextPosition = Projectile.Center + offset;
+			trail2.NextPosition = Projectile.Center + Projectile.velocity;
 		}
 
 		public void DrawPrimitives()
 		{
-			Effect effect = Terraria.Graphics.Effects.Filters.Scene["OrbitalStrikeTrail"].GetShader().Shader;
+			if (!drawTrail)
+				return;
+
+			ModContent.GetInstance<PixelationSystem>().QueueRenderAction("UnderProjectiles", () =>
+			{
+				Effect effect = Filters.Scene["CeirosRing"].GetShader().Shader;
+
+				var world = Matrix.CreateTranslation(-Main.screenPosition.Vec3());
+				Matrix view = Main.GameViewMatrix.EffectMatrix;
+				var projection = Matrix.CreateOrthographicOffCenter(0, Main.screenWidth, Main.screenHeight, 0, -1, 1);
 
-			var world = Matrix.CreateTranslation(-Main.screenPosition.Vec3());
-			Matrix view = Main.GameViewMatrix.TransformationMatrix;
-			var projection = Matrix.CreateOrthographicOffCenter(0, Main.screenWidth, Main.screenHeight, 0, -1, 1);
+				effect.Parameters["time"].SetValue(Projectile.timeLeft * -0.05f);
+				effect.Parameters["repeats"].SetValue(1);
+				effect.Parameters["transformMatrix"].SetValue(world * view * projection);
+				effect.Parameters["sampleTexture"].SetValue(Assets.GlowTrail.Value);
+
+				trail?.Render(effect);
+				trail2?.Render(effect);
+			});
+		}
+
+		private float FadeOut()
+		{
+			float opacity = 1f;
 
-			effect.Parameters["transformMatrix"].SetValue(world * view * projection);
-			effect.Parameters["sampleTexture"].SetValue(Assets.GlowTrail.Value);
-			effect.Parameters["alpha"].SetValue(1);
+			if (Projectile.timeLeft < 20)
+				opacity *= Projectile.timeLeft / 20f;
+
+			if (deathTimer > 0)
+				opacity *= deathTimer / 10f;
+
+			return opacity;
+		}
+	}
+
+	public class EarthdusterDirtShot : EarthdusterProjectile
```

Comment:
All the other multi ammo items have their projectiles in a seperate file it seems, why is this one condensed? That pattern seemed better given the large amount of small classes to be put together for the unique ammo projectiles